unityfiles/Assets/Scripts/Game/Weapon/GunController.cs

```diff
@@ -43,29 +43,34 @@ public class GunController : MonoBehaviour
 
     [Tooltip("Fire rate in shots/sec.")]
     [SerializeField] 
-    private float fireRate;  
+    private float fireRate = 1f;  
 
     // -- properties -- //
 
     public float FireRate{
         get {return fireRate;}
         set {
             this.fireRate = value;
-            this.bulletWaitTime = 1 / value;
-            // todo: it is possible for this to reference a object that does not exist
-            //          wont raise an error but wil 
-            this.fireRateTimer.Update("fRate", value);
+            this.SetBulletWaitTime(value);
+            this.fireRateTimer.Update(this.timerUID, this.bulletWaitTime);
+
         }
-    } 
+    }
 
 
     // -- internal -- //
-    private Timers fireRateTimer = new Timers();
+    private readonly Timers fireRateTimer = new Timers();
     private ArrayList activeBullets = new ArrayList();
     private ArrayList idleBullets = new ArrayList();
+
+    // REMOVE AFTER TESTING

```

Comment:
Her kan du legge inn `TODO` kode ord så det er lett å finne tilbake senere og huske å gjøre dette

Reply:
@sanderhurlen @freshfish70 @spiftire @perkynades 
Ønsker noen av dere og ha muligheten til og bare flippe skyting av og på i sidepanelet null stress og fjerne for min del man kan se for meg at dette kan være nyttig under utvikling